### DISCUSSION TITLE
How to remove picture from flutter quill document programmatically onTap method?

### DISCUSSION THREAD

**Discussion by MuhammadMusa22** [singerdmx/flutter-quill#600] [Mobile]
I have used flutter quill rich text editor in one of my project where user can create diary from flutter quill toolbar options and also can upload pictures also. So,

Q1) when user upload picture I immediately upload that picture to firesbase storage in _onImagePickCallback() method but if user hit backspace then that picture will be removed from document but it still there in firebase storage. So, how to tackle that backspace or any remove method listener inside flutter quill. (One Approach I tried is don't upload picture immediately but only at the end when user create document and want to create diary, but here I faced problem how to change all those local images to online through uploading to firebase storage.)

Q2) I have used _customEmbedBuilder() method, to draw custom design container for images in flutter quill but I also put a Cross icon on the top of picture so when user tap on that, that picture will be removed from the document regardless where the cursor is, and I don't know how to do this. Here is the link of my stack overflow question and picture of UI.
[https://stackoverflow.com/q/70729108/8805294]()

**Comment by pkozlovskiy** [comment]
Use defaultEmbedBuilder as example and try

```
                        final offset =
                            getImageNode(controller, controller.selection.start)
                                .item1;
                        controller.replaceText(offset, 1, '',
                            TextSelection.collapsed(offset: offset));
```

**Reply by MuhammadMusa22** [reply]
Thanks for your response, but there is question that this will replace image delta into text and there will be no picture displayed then,

**Reply by abcxcba** [reply]
Have you found a solution to this problem?
